### PR TITLE
Added the Gap Generator

### PR DESCRIPTION
**Pull request by abcdefg30**
It is currently invisible without a techcenter/dev cheat
as we don't have any other buildable structures in the `Defense` tab.

### CODE REVIEW

File: rules/structures.yaml [hunk 1]
```diff
@@ -399,6 +399,37 @@ gaspysat:
...
+	Building:
+		Footprint: _ x
+		Dimensions: 1,2
+	RequiresPower:
+	Health:
+		HP: 600
+	Armor:
+		Type: Wood
+	RevealsShroud:
+		Range: 10c0 # Actually 5, but needed to create 10c0 shroud
```
Comment (Phrohdoh) [inline rules/structures.yaml]: Then use 5c0.

File: rules/structures.yaml [hunk 2]
```diff
@@ -399,6 +399,37 @@ gaspysat:
...
+	CreatesShroud:
+		Range: 10c0
+	RenderShroudCircle: # TODO: Animated one
```
Comment (Phrohdoh) [inline rules/structures.yaml]: Huh? This was static.

File: sequences/structures.yaml [hunk 3]
```diff
@@ -2,42 +2,42 @@ gacnst:
 	idle: ggcnst
 		Start: 0
 		ShadowStart: 3
-		Offset: 0,-55,0
+		Offset: 0, -55
```
Comment (Phrohdoh) [inline sequences/structures.yaml]: No reason to remove the Z.

File: chrome/ingame-player.yaml [hunk 4]
```diff
@@ -267,16 +267,16 @@ Container@PLAYER_WIDGETS:
 									X: 6
 									Y: 6
 									ImageCollection: production-icons
-						ProductionTypeButton@DEFENSE:
+						ProductionTypeButton@Support:
```
Comment (Phrohdoh) [inline chrome/ingame-player.yaml]: Nit-pick: `SUPPORT`.

File: rules/player.yaml [hunk 5]
```diff
@@ -8,8 +8,8 @@ Player:
 		ReadyAudio: ConstructionComplete
 		SpeedUp: true
 		RequireOwner: false
-	ClassicProductionQueue@Defense:
-		Type: Defense
+	ClassicProductionQueue@Support:
```
Comment (Phrohdoh) [inline rules/player.yaml]: `SUPPORT`.

File: rules/structures.yaml (hunk 2 above)
Comment (abcdefg30) [inline rules/structures.yaml]: No. :)

**Comment by abcdefg30** [comment]
Updated as discussed in IRC.

**Comment by Phrohdoh** [comment]
Will merge after you test dogs. :+1:

**Pull request merged**